src/UI/RowList.elm

```diff
@@ -0,0 +1,469 @@
+            [ ( "Age", Element.text age ) ]
+    , selectMsg = Msg.ElementSelect
+    }
+
+-}
+type alias ToggleableConfig object msg =
+    ToggleableList.Config object msg
+
+
+{-| What is displayed in a collapsed toggleable-list.
+
+    { title = "Some item", caption = "Created at 2020-06-10" }
+
+-}
+type alias ToggleableCover =
+    ToggleableList.Cover
+
+
+
+-- Constructor
+
+
+{-| The main variation of `UI.RowList`.
+Click an element, and it will be selected.
+
+    RowList.selectList Msg.SelectElement
+        (\{ name } ->
+            Element.el [ Element.padding 8 ]
+                (Element.text name)
+        )
+
+-}
+selectList :
+    (object -> msg)
+    -> (RenderConfig -> Bool -> object -> Element msg)
+    -> RowList object msg
+selectList selectMsg renderItem =
+    SelectList (Properties selectMsg renderItem)
+        defaultOptions
+
+
+{-| Toggleable-lists are a variation of select-lists where the selected element expands with details while all other's details keep collapsed.
+
+We recommend using `UI.Table` instead, as it uses toggleable-lists for its responsive mode.
+
+    RowList.toggleableList
+        { detailsShowLabel = "Show details" -- For accessibility only
+        , detailsCollapseLabel = "Hide details" -- For accessibility only
+        , toCover =
+            \{ name } ->
+                { title = name, caption = Nothing }
+        , toDetails =
+            \{ age } ->
+                [ ( "Age", Element.text age ) ]
+        , selectMsg = Msg.ElementSelect
+        }
+        |> RowList.withOptions model.items
+        |> RowList.withSelected isElementSelected
+        |> RowList.renderElement renderConfig
+
+**NOTE**: Toggleable-list elements' view is not codable.
+
+-}
+toggleableList : ToggleableConfig object msg -> RowList object msg
+toggleableList config =
+    let
+        toggleableItemView parentCfg selected item =
+            if selected then
+                ToggleableList.selectedRow parentCfg config item
+
+            else
+                ToggleableList.defaultRow parentCfg config selected item
+    in
+    selectList config.selectMsg toggleableItemView
+
+
+
+-- Options
+
+
+{-| Replaces the component's action-bar.
+An action-bar is an additional pre-styled stick row that, when clicked, triggers an action.
+
+    RowList.withActionBar
+        { label = "Create new element"
+        , icon = Icon.add
+        , onClick =
+            Msg.UuidGen (Msg.ForDialog << DialogMsg.OpenElementCreation)
+        }
+        someRowList
+
+-}
+withActionBar :
+    ActionConfig msg
+    -> RowList object msg
+    -> RowList object msg
+withActionBar config (SelectList prop opt) =
+    SelectList prop { opt | actionBar = Just config }
+
+
+{-| Replaces the component's list of elements.
+
+    RowList.withOptions
+        [ { id = 0, name = "Catarina" }
+        , { id = 1, name = "Gabriel" }
+        ]
+        someRowList
+
+-}
+withOptions : List object -> RowList object msg -> RowList object msg
```

Comment:
I know it's not exactly related to this PR, but I don't think `withOptions` is a good name, we use the word `Options` internally a lot of another thing, so maybe it can be confusing. I would suggest `withItems` or `withData`